### PR TITLE
[check] Implement `fileProgress` notification for document progress

### LINKED ISSUE #54: Progress report bar
Some sort of progress bar report, or status indicating checked build. We need to investigate what support in LSP there already is for this.

### PR DESCRIPTION
**Pull request by ejgallego**
This is modelled after Lean's `$/lean/fileProgress` notification: a
yellow bar in the right side panel indicates what parts of the
document are still not checked.

Fixes #54

Note that we send on fileProgress per sentence, that's likely too
much; for now, we throttle the decoration update. Could be made into a
user-config.

### CODE REVIEW

File: controller/coq_lsp.ml [hunk 1]
```diff
@@ -179,7 +196,8 @@ let do_change params =
   let uri, version =
     (string_field "uri" document, int_field "version" document)
   in
-  Log.log_error "checking file" (uri ^ " / version: " ^ string_of_int version);
+  (* Log.log_error "checking file" (uri ^ " / version: " ^ string_of_int
```
Comment (Alizter) [inline controller/coq_lsp.ml:199]: Were you going to uncomment this?

File: examples/Rtrigo1.v [hunk 2]
```diff
@@ -1687,4 +1687,4 @@ Proof.
  intros Lo Hi [ -> | -> ].
  - now rewrite cos_PI2.
  - now rewrite cos_3PI2.
-Qed.
+Qed.
```
Comment (Alizter) [inline examples/Rtrigo1.v:1690]:
```suggestion
Qed.

```

File: examples/ex0.v [hunk 3]
```diff
@@ -99,6 +99,4 @@ Proof. by elim: m n p => //= m IHn n p; rewrite IHn. Qed.
 Lemma foobar (n : nat) : n = n.
 Proof. by []. Qed.
 
-Definition muu := 3.
-
-(* End Foo. *)
+Definition muu := 3.
```
Comment (Alizter) [inline examples/ex0.v:102]:
```suggestion
Definition muu := 3.

```

File: fleche/doc.ml [hunk 4]
```diff
@@ -23,10 +23,14 @@ type node =
   ; memo_info : string
   }
 
+(* module Contents = struct *)
```
Comment (Alizter) [inline fleche/doc.ml:26]: Remove?

**Review by Alizter — APPROVED** [review]

File: fleche/doc.ml (hunk 4 above)
Comment (ejgallego) [reply fleche/doc.ml:26]: Ups, sorry Ali, I think I messed up a rebase and included some stuff I should not; let me try again (after the 25th)

File: controller/coq_lsp.ml (hunk 1 above)
Comment (ejgallego) [reply controller/coq_lsp.ml:206]: I'm removing it, just redudant now.
